vyper/codegen/stmt.py

```diff
@@ -160,11 +160,12 @@ def parse_Call(self):
 
     def _assert_reason(self, test_expr, msg):
         if isinstance(msg, vy_ast.Name) and msg.id == "UNREACHABLE":
-            return IRnode.from_list(
-                ["assert_unreachable", test_expr], error_msg="assert unreachable"
-            )
-        elif isinstance(msg, vy_ast.Name) and msg.id == "raise UNREACHABLE":
-            return IRnode.from_list(["invalid"], error_msg="raise unreachable")
+            if test_expr is not None:
+                return IRnode.from_list(
+                    ["assert_unreachable", test_expr], error_msg="assert unreachable"
+                )
+            else:
+                return IRnode.from_list(["invalid"], error_msg="raise unreachable")
```

Comment:
nice -- since this check is performed twice, and it's actually not entirely clear what the check is for, i would factor out the expression (`test_expr is None`) into a variable called `is_raise` and add a comment about what it is doing.

Reply:
Cool! Is it possible that None can be passed into parse_Assert from an internal compiler assert? I am just thinking that if so, it may be a little misleading if it got called 'is_raise', but also it shouldn't affect behavior.

Reply:
no, `parse_Assert` only handles user asserts -- but if you are paranoid (which is not a bad thing!) you can add a sanity check in `parse_Assert` on the test expr before passing down into `assert_reason()`.